lib/identity_cache/query_api.rb

```diff
@@ -277,11 +277,12 @@ def embedded_associations
         end
       end
 
-      def cache_fetch_includes
+      def cache_fetch_includes(seen_assocs = Set.new)
         associations_for_identity_cache = recursively_embedded_associations.map do |child_association, options|
           child_class = reflect_on_association(child_association).try(:klass)
-
-          child_includes = child_class.send(:cache_fetch_includes)
+          if seen_assocs.add?(child_association)
+            child_includes = child_class.send(:cache_fetch_includes, seen_assocs)
+          end
```

Comment:
Active record includes hashes won't recursively fetch associations, so would need a lot large refactor if we chose so support it